src/main/java/nl/hsac/fitnesse/fixture/slim/mobile/AndroidAppTest.java

```diff
@@ -0,0 +1,221 @@
+
+        Dimension size = getSeleniumHelper().androidDriver().manage().window().getSize();
+        int startX = (int) (size.width * 0.2);
+        int endX = (int) (size.width * 0.8);
+        int startY = size.height / 2;
+
+        swipe(startX, startY, endX, startY, SWIPESPEED, WAITAFTERSWIPE);
+    }
+
+    public boolean swipeFrom(String direction, String locator){
+        boolean result = false;
+        MobileElement element = null;
+        if(getMobileElement(locator) != null) {
+            element = getMobileElement(locator);
+        }
+        if(element != null){
+            Dimension size = getSeleniumHelper().androidDriver().manage().window().getSize();
+            Point elementLocation = element.getLocation();
+            int startX = elementLocation.getX();
+            int startY = elementLocation.getY();
+            int endX;
+            int endY;
+
+            switch (direction.toLowerCase().charAt(0)){
+                case 'u':
+                    endX = startX;
+                    endY = 0;
+                    break;
+                case 'd':
+                    endX = startX;
+                    endY = size.getHeight();
+                    break;
+                case 'l':
+                    endX = 0;
+                    endY = startY;
+                    break;
+                case 'r':
+                    endX = size.getWidth();
+                    endY = startY;
+                    break;
+                default:
+                   System.err.print("ERROR: " + direction + " is not a valid swipe direction.");
+                   return false;
+            }
+            swipe(startX, startY, endX, endY, SWIPESPEED, WAITAFTERSWIPE);
+            result = true;
+        }
+        else{
+            System.err.print("ERROR: Element not found for " + locator);
+        }
+        return result;
+    }
+
+    private void swipe(int startX, int startY, int endX, int endY, int speed, int waitAfter){
+        getSeleniumHelper().androidDriver().swipe(startX, startY, endX, endY, speed);
+        waitMilliseconds(waitAfter);
+    }
+
+    public void pressBackButton(){
+        getSeleniumHelper().androidDriver().navigate().back();
+    }
+
+    private MobileElement getMobileElement(String locator){
+        String lookFor;
+        String content;
+
+        AndroidDriver<MobileElement> driver = getSeleniumHelper().androidDriver();
+
+        //assume text lookup if none is specified (convenient)
+        if(!locator.contains("=")){
+            lookFor = "text";
+            content = locator;
+        }
+        //Otherwise, use specified UiSelector method
+        else{
+            String[] strategy = locator.split("=");
+            lookFor = strategy[0];
+            content = strategy[1];
+        }
+
+        return driver.findElementByAndroidUIAutomator("new UiSelector()." + lookFor + "(\"" + content + "\")");
+
+    }
+
+    @Override
+    public String takeScreenshot(String basename) {
```

Comment:
As above. I needed the helperclass to know what class was asking for the screenshot. That definitely needs work.